tests/components/airpatrol/test_climate.py

```diff
@@ -0,0 +1,424 @@
+"""Test the AirPatrol climate platform."""
+
+from datetime import timedelta
+
+from freezegun.api import FrozenDateTimeFactory
+
+from homeassistant.components.airpatrol.climate import (
+    HA_TO_AP_FAN_MODES,
+    HA_TO_AP_HVAC_MODES,
+    HA_TO_AP_SWING_MODES,
+)
+from homeassistant.components.climate import (
+    ATTR_CURRENT_HUMIDITY,
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_FAN_MODE,
+    ATTR_FAN_MODES,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_SWING_MODE,
+    ATTR_SWING_MODES,
+    DOMAIN as CLIMATE_DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_SWING_MODE,
+    SERVICE_SET_TEMPERATURE,
+    SWING_OFF,
+    SWING_ON,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, SERVICE_TURN_OFF, SERVICE_TURN_ON
+from homeassistant.core import HomeAssistant
+
+from tests.common import async_fire_time_changed
+
+
+async def test_async_setup_entry_adds_entities(
+    hass: HomeAssistant, load_integration, mock_api_response, get_data
+) -> None:
+    """Test async_setup_entry creates and adds AirPatrolClimate entities that have climate data."""
+    unit_with_climate = get_data(unit_id="unit1", name="Unit 1")
+    unit_without_climate = get_data(unit_id="unit2", name="Unit 2", climate=None)
+    mock_api_response.return_value = [
+        unit_with_climate[0],
+        unit_without_climate[0],
+    ]
+    await load_integration()
+
+    state = hass.states.get("climate.unit_1")
+    assert state
+
+    state = hass.states.get("climate.unit_2")
+    assert state is None
```

Comment:
You can use the `snapshot_platform` helper here. If you add that you can do `pytest ./tests/components/airpatrol --snapshot-update` and it will fixate the entity in a .ambr file